### PR TITLE
we should display correct answer in green when showInstantfeedback is true

### PR DESCRIPTION
**Pull request by lvdatri1**
we should display correct answer in green when showInstantfeedback is true
I just added small changes to add "correct" class to correct button if they select wrong answer, and the answer is true.

### CODE REVIEW

File: src/docs/index.jsx [hunk 1]
```diff
@@ -15,7 +15,7 @@ function App() {
         quiz={quiz}
         shuffle
         shuffleAnswer
-      // showInstantFeedback
+      showInstantFeedback
```
Comment (wingkwong) [inline src/docs/index.jsx:18]: please fix the indentation

File: src/docs/quiz.jsx [hunk 2]
```diff
@@ -7,7 +7,7 @@ const segment = {
 const quiz = {
   quizTitle: 'React Quiz Component Demo',
   quizSynopsis: 'Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Donec quam felis, ultricies nec, pellentesque eu, pretium quis, sem. Nulla consequat massa quis enim',
-  nrOfQuestions: '6',
+  nrOfQuestions: '2',
```
Comment (wingkwong) [inline src/docs/quiz.jsx:10]: revert this change please

File: src/lib/Core.jsx [hunk 3]
```diff
@@ -255,9 +255,10 @@ function Core({
             <button
               type="button"
               disabled={answerButtons[index].disabled || false}
-              className={`${answerButtons[index].className} answerBtn btn`}
+              className={`${answerButtons[index].className} answerBtn btn ${(index+1==correctAnswer&&showInstantFeedback)?'correct':''}`}
```
Comment (wingkwong) [inline src/lib/Core.jsx:258]: lint once

```jsx
className={`${answerButtons[index].className} answerBtn btn ${(index + 1 == correctAnswer && showInstantFeedback) ? 'correct' : ''}`}
```

File: src/lib/Core.jsx [hunk 4]
```diff
@@ -255,9 +255,10 @@ function Core({
...
               onClick={() => (revealAnswerOnSubmit ? onSelectAnswer(index) : onClickAnswer(index))}
             >
+              {/* {<div>index +{index}and correct answer is:{correctAnswer}</div>} */}
```
Comment (wingkwong) [inline src/lib/Core.jsx:261]: can be removed

File: src/lib/Core.jsx [hunk 5]
```diff
@@ -268,6 +269,8 @@ function Core({
               onClick={() => (revealAnswerOnSubmit ? onSelectAnswer(index) : onClickAnswer(index))}
               className={`answerBtn btn ${(allowNavigation && checkSelectedAnswer(index + 1)) ? 'selected' : null}`}
             >
+{/* {<div>before Trinh selelct</div>} */}
+
```
Comment (wingkwong) [inline src/lib/Core.jsx:273]: can be removed

**Review by wingkwong — CHANGES_REQUESTED** [review]
1. I'm expecting two lines of changes in this PR (1 from `src/lib/Core.jsx` and 1 from `src/docs/index.jsx`).
2. You don't need to commit `dist/*.js`. These files will be built during the pre-release.

File: src/lib/Core.jsx (hunk 3 above)
Comment (chyke007) [reply src/lib/Core.jsx:258]: className={`${answerButtons[index].className} answerBtn btn ${(index + 1 === Number(correctAnswer) && showInstantFeedback) ? 'correct' : ''}`}

Added Number(correctAnswer) as I observed from testing locally that correctAnswer was a string, then the check failed

**Comment by chyke007** [comment]
Is this still worked on, so as to be merged in?

File: src/lib/Core.jsx (hunk 3 above)
Comment (wingkwong) [reply src/lib/Core.jsx:258]: @chyke007 Thanks for pointing out. Apart from that, I think there are two cases for `correctAnswer`. 

See

https://github.com/wingkwong/react-quiz-component/blob/04b242b765c040a079bf4f8d70f43bae578bd7ca/src/lib/Core.jsx#L123-L131

**Comment by wingkwong** [comment]
@chyke007 I don't think he's still working on it. If you're interested, you may take over it.

File: src/lib/Core.jsx (hunk 3 above)
Comment (chyke007) [reply src/lib/Core.jsx:258]: Thats correct also, that logic wont work for arrays

**Comment by chyke007** [comment]
> @chyke007 I don't think he's still working on it. If you're interested, you may take over it.

Currently trying how to make the array part work, once i figure that out, would create a PR

**Comment by chyke007** [comment]
> we should display correct answer in green when showInstantfeedback is true

@wingkwong created a PR for this [here](https://github.com/wingkwong/react-quiz-component/pull/188)

**Comment by chyke007** [comment]
> @chyke007 I don't think he's still working on it. If you're interested, you may take over it.

@wingkwong created a PR for this [here](https://github.com/wingkwong/react-quiz-component/pull/188)

**Comment by wingkwong** [comment]
closing due to inactivity. will be followed up in Chibuike's PR.
